numpy/core/src/multiarray/sequence.c

```diff
@@ -69,22 +69,66 @@ NPY_NO_EXPORT PySequenceMethods array_as_sequence = {
 static int
 array_any_nonzero(PyArrayObject *arr)
 {
-    npy_intp counter;
-    PyArrayIterObject *it;
-    npy_bool anyTRUE = NPY_FALSE;
+    PyArray_NonzeroFunc *nonzero;
```

Comment:
this function is only called from `array_contains` which should ensure the input is a boolean array. So we should just be able to use `count_boolean_trues`. While that function does not early abort it is so heavily optimized that it is probably still faster for many cases and its performance is irrelevant compare to the richcompare earlier

Reply:
just doing PyArray_CountNonZero() > 0` gives these timings for 100000 int64 elements

```
original:
slow exit: 341
fast exit: 41us

using count_nonzero:
46.1us
46.1us
```

So slightly slower for fast exits but it removes some non-trivial code so I think it is worthwhile.
If we want to keep the early exit as is one could add a fast exit option to `count_boolean_trues`

Reply:
We could also call `PyArray_Any`, which seems even more literal. Then `array_contains` would be exactly equivalent to `(a == val).any()`.

`PyArray_Any` doesn't short-circuit either, but if anyone makes it do so in the future then we would get the speed benefit for free in `array_contains`.